adoc/MAIN.release-notes.adoc

```diff
@@ -19,6 +19,65 @@ Release notes usually only list changes that happened between two subsequent rel
 Certain important entries from the release notes documents of previous product versions may be repeated.
 To make such entries easier to identify, they contain a note to that effect.
 
+== Changes in 4.1.1
+
+{productname} can now be deployed with {tf} 0.12.
+
+=== Required Actions
+
+==== Upgrade terraform files and state
+
+If you can tear down your existing cluster, delete your cluster before
+upgrading to {tf} 0.12. You can then create a cluster following our
+documentation. It is the cleanest path.
+
+If you are using {tf} 0.11 and you cannot tear down your cluster, you will
+need to update your {tf} files (and states) in place for {tf} 0.12
+
+For this, enter your {tf} files/state folder and:
+
+* Migrate {tf} files with the automatic migration tool:
+  ``terraform 0.12upgrade``.
+
+==== Extra operations for in-place upgrade of libvirt terraform files
+
+* Replace any boolean written as a number to false/true. For example, replace
+  ``wait_for_lease = 1 `` with ``wait_for_lease = true`` in
+  ``libvirt/{lb|master|worker}-instance.tf``
+
+===== Extra operations for in-place upgrade of OpenStack terraform files
+
+* Replace any boolean written as a number to false/true. For example, replace
+  ``default = 0`` with ``default = false`` in ``openstack/variables.tf``.
+* Introduce a new variable ``password`` in ``variables.tf``.
+* Introduce a ``depends_on`` on the resource ``"openstack_compute_floatingip_associate_v2" "master_ext_ip"``:
+  ``
+  depends_on = [openstack_compute_instance_v2.master]
+  ``
+* Introduce a ``depends_on`` on the resource ``"master_wait_cloudinit"``:
+  ``
+  depends_on = [
+    openstack_compute_instance_v2.master,
+    openstack_compute_floatingip_associate_v2.master_ext_ip
+  ]
+  ``
+* Introduce a ``depends_on`` on the resources
+  ``"openstack_compute_floatingip_associate_v2" "worker_ext_ip"`` and
+  ``"null_resource" "worker_wait_cloudinit"``, similarly to the ones for master.
+  Replace ``master`` with ``worker`` on the examples above.
+
+* Update the resources ``resource "openstack_compute_instance_v2" "master"``
+  and ``resource "openstack_compute_instance_v2" "worker"`` in respectively
+  ``master-instance.tf`` and ``worker-instance.tf``. Add in those resources:
+  ``
+  lifecycle {
+    ignore_changes = [user_data]
```

Comment:
I think this change is dangerous because will prevent updates of nodes if we ever change the userdata.

Reply:
Correct. It's dangerous. Should we make it clearer?

Reply:
Could you check if the latest version is good enough for you?

Reply:
afaik this is only for the update of the files no? should this be removed inmediatly after upgrading 0.11 to 0.12? or doing that means that your cluster will be recreated on the next terraform action?

Not clear to me if this is forever or temporary

Reply:
I think that if you remove it after the upgrade, the state machine would still have different user_data, which will lead to a complete teardown of the cluster. So yes it's a little bit forever.

@bergmannf can you confirm? I didn't test the multiple update scenario.